### ISSUE TITLE
How to refresh the authCode on request without signOut

### ISSUE THREAD

**Issue by supersolid-Justy** [playgameservices/play-games-plugin-for-unity#1821]
Hello!

I am using the older version of the plugin (0.9.36) in my client app, but will need to update it to the newer versions.
I see that the GetServerAuthCode() has been changed to not accept a callback any more and the underlying code also retrieves the authCode that is saved from when the user signed in, so it seems to be kept within the app session, as long as the user doesnt log out and logs back in again.
The server code used for the app is however expecting a one-time-only authCode, which does not seem to be supported any more.
From the website linked in the main page:
https://developers.google.com/identity/sign-in/web/backend-auth
I see that the one-time-only authCode is still used by the server, so I was wondering if that is a bug in the plugin, or if the client should sending a authCode that stays the same throughout a login session and its the server has to generate a one-time-only code?
Thank you very much in advance!

**Comment by mkorotetsky** [comment]
Hi, I don't understand why it was changed. I need to refresh authCode too. For now I added some hacks to code to force refresh it manually but I think it is not stable and crashes sometimes.

**Comment by supersolid-Justy** [comment]
Still waiting for a response. I currently am writing the one-time server auth code retrieval outside of the authentication process back into the plugin and might be asking for a pull request to add that in as it seems like some people still need to get a new server auth code on demand.
How our application currently works is that we call GetServerAuthCode() several times when the user is logged in to google play and wants to back up progress or switch from one application progress to another. The first time is to see if the user has already a progress that is on the server, but not on the client, the second one is to retrieve the backed up progress while ensuring its still the same player. We do not need to store any access/refresh tokens on the server this way.
It seems to prevent issues such as the one described in here:
https://github.com/playgameservices/play-games-plugin-for-unity/issues/1804

https://github.com/playgameservices/play-games-plugin-for-unity/issues/984 (and old threat that is closed already) it seems that the way the plugin was meant to work was that getServerAuthCode() should only be called once? However it was possible to call GetServerAuthCode() every time you talked to the google API, which is how we used it. I guess we did it wrong in the first place then? But I also am wondering why asking for the one-time only auth code on demand is not supported any more on the plugin side and hope that Mr Wilkinson could share some more light about it :)!
